File: sentimental-credit/credit.py
from cs50 import get_int


def main():
    card_number = get_cardnumber()
    card_length = get_cardlength(card_number)
    card_type = get_cardtype(card_number)
    card_mod = get_cardmod(card_number)
    check_card(card_length, card_type, card_mod)


def get_cardnumber():
    while (True):
        card_number = get_int("Number: ")
        if card_number >= 0:
            return card_number


def get_cardlength(card_number):
    card_length = len(str(card_number))
    return card_length


def get_cardtype(card_number):
    while card_number > 99:
        card_number = card_number // 10
    return card_number


def get_cardmod(card_number):
    remainder = 0
    sum = 0
    while card_number > 0:
        remainder = remainder + (card_number % 10)
        card_number = card_number // 10
        luhnmod = (card_number % 10) * 2
        if luhnmod > 9:
            sum = sum + (luhnmod - 9)
        else:
            sum = sum + luhnmod
        card_number = card_number // 10
    card_mod = (remainder + sum) % 10
    return card_mod


def check_card(card_length, card_type, card_mod):
    if card_mod == 0:
        if (card_type == 34 or card_type == 37) and card_length == 15:
            print("AMEX")
        elif (card_type >= 51 and card_type <= 55) and card_length == 16:
            print("MASTERCARD")
        elif (card_type >= 40 and card_type <= 49) and (card_length == 13 or card_length == 16):
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")


main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long cardNumber;
    long cardType;
    long counter;
    int sum;
    int remainder;
    int luhnMod;
    int cardLength;
    int totalMod;

    do
    {
        cardNumber = get_long("Number: ");
    }
    while (cardNumber < 0);

    cardType = cardNumber;

    while (cardType > 99)
    {
        cardType = cardType / 10;
    }

    counter = cardNumber;

    while (counter > 0)
    {
        cardLength = cardLength + 1;
        counter = counter / 10;
    }

    while (cardNumber > 0)
    {
        remainder = remainder + (cardNumber % 10);
        cardNumber = cardNumber / 10;
        luhnMod = (cardNumber % 10) * 2;
        if (luhnMod > 9)
        {
            sum = sum + (luhnMod - 9);
        }
        else
        {
            sum = sum + luhnMod;
        }
        cardNumber = cardNumber / 10;
    }

    totalMod = (remainder + sum) % 10;

    if (totalMod == 0)
    {
        if ((cardType == 34 || cardType == 37) && cardLength == 15)
        {
            printf("AMEX\n");
        }
        else if ((cardType >= 51 && cardType <= 55) && cardLength == 16)
        {
            printf("MASTERCARD\n");
        }
        else if ((cardType >= 40 && cardType <= 49) && (cardLength == 13 || cardLength == 16))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}